serdaSimus/assignments/twopointers/lc26/lc26.js

```diff
@@ -0,0 +1,114 @@
+// Problem Statement #
+// Given an array of sorted numbers, remove all duplicates from it. You should not use any extra space; after removing the duplicates in-place return the length of the subarray that has no duplicate in it.
+
+/*
+  SOLUTION #1
+	input: arr: unsorted array.
+	input2: key:interger
+
+	setup:
+		ptr to start of arr.
+		ptr to end of arr.
+		swap_mechanism
+	
+	process:
+		while loop(ptrStar <= ptrEnd):
+			if(elemnt[end]== key)
+				move ptrEnd to -1
+			else:
+				if(elem[start] == key):
+					swap(elem[start] with elem[end])
+					pop()
+					move (startPtr by 1)
+					move (endPtr by -1)
+				else
+					move startPtr by +
+
+*/
+
+// In-class, group solution Serda & Tiantian
+// Swap() and pop().
+function remove_element(arr, key) {
+  // TODO: Write code here
+  if (!arr.length) return null;
+
+  let windowStart = 0;
+  let windowEnd = arr.length - 1;
+
+  // Tc: O(n) where n is the size of the array.
+  // Sc: O(3) which essentially is O(1) or constant space.
+  while (windowStart <= windowEnd) {
+    if (arr[windowEnd] === key) {
+      windowEnd--;
+      arr.pop();
+    } else {
+      if (arr[windowStart] == key) {
+        let temp = arr[windowStart]; //O(1)
+        arr[windowStart] = arr[windowEnd];
+        arr[windowEnd] = temp;
+
+        windowEnd--;
+        arr.pop();
+        windowStart++;
+      } else {
+        windowStart++;
+      }
+    }
+  }
+  return arr.length;
+}
```

Comment:
This is a great joined effort! I would recommend afterwards to take a look at the solution provided/on Leetcode to see how to make the code more efficient, and then write it out so you build that muscle memory!